services/core/InfluxdbHistorian/influx/historian.py

```diff
@@ -0,0 +1,396 @@
+    utils.update_kwargs_with_config(kwargs, config_dict)
+    _log.debug("In influx historian before calling class kwargs is {}".format(
+        kwargs))
+    return InfluxdbHistorian(connection, aggregations, **kwargs)
+
+
+class InfluxdbHistorian(BaseHistorian):
+    """
+        Historian that stores the data into influxdb client's database
+    """
+
+    def __init__(self, connection, aggregations, **kwargs):
+        """
+        Initialise the historian.
+
+        :param connection: dictionary that contains necessary information to
+        establish a connection to the influxdb database. The dictionary should
+        contain entry 'params', which contains:
+            1. host
+            2. port
+            3. database
+            4. user
+            5. passwd
+        :param kwargs: additional keyword arguments. (optional identity and
+                       topic_replace_list used by parent classes)
+        """
+        super(InfluxdbHistorian, self).__init__(**kwargs)
+        self._connection_params = connection.get('params', {})
+        self._host = self._connection_params.get('host', None)
+        self._user = self._connection_params.get('user', None)
+        self._database = self._connection_params.get('database', None)
+        self._client = None
+
+        # Config for aggregation queries, can be changed in config file.
+        self._use_calendar_time_periods = aggregations.get('use_calendar_time_periods', False)
+
+        config = {
+            "connection": connection,
+            "aggregations": aggregations
+        }
+
+        self.update_default_config(config)
+
+        # meta_dicts to keep track of meta dictionary for all topics.
+        self._meta_dicts = {}
+        self._topic_id_map = {}
+
+    def configure(self, configuration):
+        """
+        The expectation that configuration will have at least the following
+        items if authentication is enabled in InfluxDB.
+
+        .. code-block:: python
+
+            {
+              "connection": {
+                "params": {
+                  "host": "localhost",
+                  "port": 8086,
+                  "database": "historian",
+                  "user": "historian",
+                  "passwd": "historian"
+                }
+              }
+            }
+
+        If user and passwd are optional if authentication is disabled
+        """
+        try:
+            params = configuration['connection']['params']
+            host = params['host']
+            db = params['database']
+            user = params.get('user', None)
+            passwd = params.get('passwd', None)
+            if configuration['aggregations']:
+                use_calendar_time_periods = configuration['aggregations']['use_calendar_time_periods']
+        except (KeyError, TypeError) as err:
+            _log.error('Invalid configuration: %s', err)
+            raise err
+
+        if host is None:
```

Comment:
probably should just use 
```` python
if not host:
````
to protect against empty strings...
Same to db etc.